tests/test_measurements.py

```diff
@@ -524,6 +526,134 @@ def test_shape_shot_vector_no_obs(self):
             res.shape(dev)
 
 
+class TestCounts:
+    """Tests for the counts function"""
+
+    def test_counts_dimension(self, tol):
+        """Test that the sample function outputs counts of the right size"""
+        n_sample = 10
+
+        dev = qml.device("default.qubit", wires=2, shots=n_sample)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.54, wires=0)
+            return qml.sample(qml.PauliZ(0), counts=True), qml.sample(qml.PauliX(1), counts=True)
+
+        sample = circuit()
+
+        assert np.array_equal(sample.shape, (2,)) & np.all(
+            [sum(s.unwrap().values()) == n_sample for s in sample]
+        )
```

Comment:
It would be nice to break down this line into two separate `assert` statements to enhance readability.

Reply:
done in [this commit](https://github.com/PennyLaneAI/pennylane/pull/2686/commits/b9dc8ef4d3f3c643a4d2472aaf74db80ef670013)